OneSignalSDK/onesignal/src/main/java/com/onesignal/JSONUtils.java

```diff
@@ -49,7 +51,10 @@ static JSONObject generateJsonDiff(JSONObject cur, JSONObject changedTo, JSONObj
                         JSONObject outValue = null;
                         if (baseOutput != null && baseOutput.has(key))
                             outValue = baseOutput.getJSONObject(key);
-                        JSONObject returnedJson = generateJsonDiff(curValue, (JSONObject) value, outValue, includeFields);
+                        JSONObject returnedJson;
+                        synchronized (syncLock) {
+                            returnedJson = generateJsonDiff(curValue, (JSONObject) value, outValue, includeFields);
```

Comment:
Just curious, why do we synchronize this line and not the method?

Reply:
This function is already called syncronized when used, but I noticed the exception is thrown when called recursively, so it was necessary to synced call this as well